pkg/query_expr/es_expr.go

```diff
@@ -0,0 +1,279 @@
+package queryexpr
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/expr-lang/expr/ast"
+)
+
+type ESExpr string
+
+func (e *ESExpr) String() string {
+	return string(*e)
+}
+
+// ToQuery default
+func (e *ESExpr) ToQuery() (string, error) {
+	queryExprParsed, err := GetTreeNodeFromQueryExpr(e.String())
+	if err != nil {
+		return "", err
+	}
+
+	esQueryInterface, err := e.translateToEsQuery(queryExprParsed)
+	if err != nil {
+		return "", err
+	}
+	esQuery, ok := esQueryInterface.(map[string]interface{})
+	if !ok {
+		return "", fmt.Errorf("failed to generate Elasticsearch query")
+	}
+	esQuery = map[string]interface{}{"query": esQuery}
+
+	// Convert to JSON
+	queryJSON, err := json.Marshal(esQuery)
+	if err != nil {
+		return "", err
+	}
+
+	return string(queryJSON), nil
+}
+
+// Validate default: no validation
+func (*ESExpr) Validate() error {
+	return nil
+}
+
+// translateToEsQuery The idea came from ast.Walk. Currently, the development focus implement for the node type that most likely used in our needs.
+// TODO: implement translator for node type that still not covered right now.
+func (e *ESExpr) translateToEsQuery(node ast.Node) (interface{}, error) {
+	if node == nil {
+		return nil, fmt.Errorf("cannot convert nil to Elasticsearch query")
+	}
+	switch n := (node).(type) {
```

Comment:
do we need all the cases below? or just some?

Reply:
All cases below needed, depends on query expr. The current that I handle are all cases that can happened right now.

I already note this too to our client in our docs: https://github.com/goto/compass/pull/78/files